thorlcr/graph/thgraphslave.cpp

```diff
@@ -140,10 +140,29 @@ void CSlaveActivity::setInput(unsigned index, CActivityBase *inputActivity, unsi
     else
         outLink.set(((CSlaveActivity *)inputActivity)->queryOutput(inputOutIdx));
     assertex(outLink);
+
     while (inputs.ordinality()<=index) inputs.append(NULL);
+
     inputs.replace(outLink.getClear(), index);
 }
 
+void CSlaveActivity::appendOutput(IThorDataLink *itdl)
+{
+    if (queryJob().getOptBool("TRACEROWS"))
+    {
+        const unsigned numTraceRows = queryJob().getOptInt("numTraceRows", 10);
+        outputs.append(new CTracingThorDataLink(itdl, queryHelper(), numTraceRows));
```

Comment:
I don't think there is (or was) any casting to the implemented class, except NSplitter's special case where it was casting to it's own derivative of CThorDataLink, which as @shamser says has been handled differently now (not using queryOutput()).

If there were any miscasting, you'd likely get a crash than missing meta info.

@GordonSmith, is the missing meta info involving particular activities/edges ?